frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-page.tsx

```diff
@@ -90,27 +90,37 @@ const InstallCluster: React.FC<InstallClusterProps> = ({ match }) => {
     return sanitizedMode;
   };
 
-  const getStep = (offset: number = 0) => {
-    const searchParams = new URLSearchParams(window.location.search.slice(1));
-    const step = parseInt(searchParams.get('step'), 10) || 1;
-    const sanitizedStep = step && step <= 5 - offset && step >= 1 ? step : 1;
-    return sanitizedStep;
-  };
-
   const getParamString = (step: number, mode: number) => {
     const searchParams = new URLSearchParams(window.location.search.slice(1));
     searchParams.set('step', step.toString());
     searchParams.set('mode', mode.toString());
     return searchParams.toString();
   };
 
+  const getAnchor = (step: number, mode: number) => `~new?${getParamString(step, mode)}`;
+
+  const getStep = (
+    offset: number = 0,
+    forceNavigate: boolean = false,
+    isForceNavigated: boolean = false,
+    setForceNavigated: React.Dispatch<React.SetStateAction<boolean>> = null,
+  ) => {
+    const searchParams = new URLSearchParams(window.location.search.slice(1));
+    const step = parseInt(searchParams.get('step'), 10) || 1;
+    let sanitizedStep = step && step <= 5 - offset && step >= 1 ? step : 1;
+    if (forceNavigate && !isForceNavigated) {
+      history.push(getAnchor(3, 2));
+      sanitizedStep = 3;
+      setForceNavigated(true);
+    }
+    return sanitizedStep;
+  };
+
   const getIndex = (searchSpace: any, search: string, offset: number = 0) => {
     const index = Object.values(searchSpace).findIndex((el) => el === search);
     return index - offset + 1;
   };
 
-  const getAnchor = (step: number, mode: number) => `~new?${getParamString(step, mode)}`;
-
   const handleModeChange = (event: React.FormEvent<HTMLInputElement>) => {
```

Comment:
IMO this is the function that should take care of setting up the step correctly for Internal Attached Mode. Why can't we make changes here? In cases where `modeIndex` is 2 based on the presence of a no-prov sc change the parameters you pass to `getIndex` at line 118.

Reply:
Actually ,`handleModeChange` is executed before `getStep`, since `getStep` is linked with the [Wizard component render](https://github.com/rexagod/console/blob/5a7193a28523fc10468e711a3c3bd25f07e70ff1/frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices-mode/install-wizard.tsx#L304), hence I used the `getStep` one.

Reply:
My suggestion is that you setup the step and mode in the URL in the above function. And let `getStep` parse the URL and set the correct step.

Reply:
```
Actually ,handleModeChange is executed before getStep
```
This actually makes this a better approach as you can use `handleModeChange` to contain logic related to setting up unique logic for each mode.

Reply:
Seconded, that's a way better approach. On it.